### ISSUE TITLE
Input composition: Korean letters can be heard in edit fields in apps where NVDA is told to sleep for a while

### ISSUE THREAD

**Issue by josephsl** [nvaccess/nvda#7208]
Hi,

A Korean contributor reports:

If using an app with NVDA put to sleep (NVDA+Shift+S) and when entering Korean letters into an edit field, NVDA will announce Korean letters.

STR:

1. With Korean IME installed, open Notepad.
2. Type an English sentence.
3. Type some Korean letters (such as "rkskek" with Korean IME set to native input).
4. Repeat steps 2 and 3 but with NVDA told to sleep.

Expected: NVDA does not announce Korean letters as one types them.
Actual: English letters are not announced while Korean letters are despite NVDA sleeping.

I was able to confirm this. Thanks.

**Comment by jcsteh** [comment]
The functions in NVDAHelper.py that receive/handle stuff related to input methods don't currently check for sleepMode. They need to be tweaked to do so.
